### ISSUE TITLE
Document overriding nuget.exe path with v1.* task

### ISSUE THREAD

**Issue by ljani** [Microsoft/vsts-tasks#5219]
Currently there is no public documentation how to override `nuget.exe` path with the `v1.*` restore task. The related changeset is [`2b50ca`](https://github.com/Microsoft/vsts-tasks/commit/2b50ca67d6f8347ebfaf6f5b940b345d97411392). (Thank you for open sourcing your stuff.)

Overriding nuget path is currently needed to fix eg. #5164 to upgrade `nuget.exe` with on-premises TFS 2017 Update 2. The tool infrastructure seems to be chaging this though.

TL;DR, use `NuGetExeToolPath` environment variable to configure path to `nuget.exe`.

**Comment by keithrob** [comment]
@ljani ,
The documented way to override nuget.exe is to use the nuget tool installer task ahead of any subsequent build steps.  This will place the version of nuget you selected from the tool installer at the front of $PATH.  Notice [this](https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/Common/nuget-task-common/NuGetToolGetter.ts#L88).

Public documentation on the nuget tool installer is [here](https://docs.microsoft.com/en-us/vsts/build-release/tasks/tool/nuget).
Keith

**Comment by ljani** [comment]
That task is only going to land in TFS 2018, right?

**Comment by keithrob** [comment]
Ah yes, I missed the 2017U2.  My concern; however, is that you're using an internal variable that could change and I don't want your builds to break should we rename it or something.

You may want to revert back to the earlier version of the task which has the ability to provide a path to nuget.exe.

**Comment by ljani** [comment]
> My concern; however, is that you're using an internal variable that could change and I don't want your builds to break should we rename it or something.

Ah, thanks for clarification.
